allennlp/models/encoder_decoders/nlvr_semantic_parser.py

```diff
@@ -788,16 +796,18 @@ def __init__(self,
                  encoder_output_dim: int,
                  action_embedding_dim: int,
                  attention_function: SimilarityFunction,
-                 checklist_weight: float) -> None:
+                 num_terminals: int) -> None:
         """
         Parameters
         ----------
         encoder_output_dim : ``int``
         action_embedding_dim : ``int``
         attention_function : ``SimilarityFunction``
-        checklist_weight : ``float`` (optional)
-            Weight assigned to checklist score when computing the final score. The final score will
-            be ``checklist_weight * checklist_score + (1 - checklist_weight) * action_prob``.
+        num_terminals : ``int``
```

Comment:
You should call this `checklist_size`, not `num_terminals`.  In all of your documentation, you're explaining why `num_terminals` is needed, or just referring to a `checklist_size` directly - much simpler if you just call this what it is.